### PR TITLE
Added documentation for reindexing.

### PR DESCRIPTION
**Pull request by dantleech**
Added some documentation on reindexing and ways to improve indexing speed.

### CODE REVIEW

File: bundles/search.rst [hunk 1]
```diff
@@ -99,10 +99,49 @@ index, and can modify it, before it is compared with the first argument from
...
+the metadata and it is desirable to propagate these changes over all of the
+indexed documents / entities in the system -- or when you import new data
+(e.g. from a backup) and need to index that data.
+
+To reindex all entities (Contacts, Media, etc) and documents (Pages, Snippets)
+simply run the following:
+
+.. code-block:: bash
+
+    $ ./app/console massive:search:index:rebuild --env=prod
```
Comment (danrot) [inline bundles/search.rst]: Don't you want to use the new `reindex` command here already?

Comment (dantleech) [inline bundles/search.rst]: Yeah, was written before that.

Comment (dantleech) [inline bundles/search.rst]: Updated.

**Comment by webmozart** [comment]
Looks good to me. :)

File: bundles/search.rst [hunk 2]
```diff
@@ -99,10 +99,49 @@ index, and can modify it, before it is compared with the first argument from
...
+  reduce logging and increase speed and lead to lower memory consumption.
+- Ensure that the document manager has the ``debug: false`` option. This
+  reduces logging dramatically.
+- Use `PHP 7`_: This will increase indexing speed up to 10 times.
+
+To recover if the process is interrupted:
+
+- You may *resume* the task simply by running it again.
+- Use the ``--filter`` option to filter which entity/document classes are
+  re-indexed, e.g. ``--filter=Contact``.
```
Comment (dantleech) [inline bundles/search.rst]: `--filter` is no longer supported with the latest refactorings ..

Comment (danrot) [inline bundles/search.rst]: Can you update it then? :-)

**Pull request merged**